challenge-manager/chain-watcher/watcher.go

```diff
@@ -585,6 +586,18 @@ func (w *Watcher) AddEdge(ctx context.Context, edge protocol.SpecEdge) (bool, er
 	if isRoyalEdge {
 		srvlog.Info("Observed an honest challenge edge created onchain, now tracking it locally", fields)
 	} else {
+		if edge.ClaimId().IsSome() {
+			evilEdges, ok := w.evilEdgesByLevel.TryGet(edge.GetChallengeLevel())
+			if !ok {
+				evilEdges = threadsafe.NewSet[protocol.EdgeId](threadsafe.SetWithMetric[protocol.EdgeId]("evilEdges"))
+				w.evilEdgesByLevel.Put(edge.GetChallengeLevel(), evilEdges)
+			}
+			evilEdges.Insert(edge.Id())
+			if evilEdges.NumItems() > 5 {
```

Comment:
We should avoid inserting more into the map here once we hit 5

Reply:
done.